src/os/bluestore/BlueStore.h

```diff
@@ -1177,6 +1177,45 @@ class BlueStore : public ObjectStore,
     }
   }
   CompressionMode comp_mode = COMP_NONE;      ///< compression mode
+
+  // compression algorithm
+  enum CompressionAlgorithm {
+    COMP_ALG_NONE,
+    COMP_ALG_SNAPPY,
+    COMP_ALG_ZLIB,
+  };
+
+  string get_comp_alg_name(int a) {
+    string alg;
+
+    switch (a) {
+    case COMP_ALG_NONE:
+      break;
+    case COMP_ALG_SNAPPY:
+      alg = "snappy";
+      break;
+    case COMP_ALG_ZLIB:
+      alg = "zlib";
+      break;
+    default:
+      break;
+    }
+
+    return alg;
+  }
+
```

Comment:
I think these all belong in bluestoer_types.h and cc, probably along side CSUM_*